### ISSUE TITLE
Missing HTML pages and broken links for undocumented C++ concepts

### ISSUE THREAD

**Issue by jwakely** [doxygen/doxygen#9682]
**Describe the bug**

When you have a C++ concept that has no documentation comment, Doxygen generates links to HTML pages that don't exist, resulting in 404 errors.

**Expected behavior**

Either generate a plain documentation page just showing the concept name and template params, or don't create a link for it.

**To Reproduce**

```
mkdir tmp
cd tmp
cat > foo.h <<EOT
/// The FOO concept
template<typename T> concept foo = true;

template<typename T> concept bar = true;
EOT
doxygen .
```

Then view the `html/index.html` page, click on "Concepts" in the top menu, and then click on "bar".  The link is to a page called `conceptbar.html` but no such page was created:

```
$ ls html
bc_s.png         dynsections.js      menudata.js   sync_on.png
bdwn.png         files.html          nav_f.png     tab_a.png
closed.png       folderclosed.png    nav_g.png     tab_b.png
conceptfoo.html  folderopen.png      nav_h.png     tab_h.png
concepts.html    foo_8h_source.html  open.png      tab_s.png
doc.png          index.html          search        tabs.css
doxygen.css      jquery.js           splitbar.png
doxygen.svg      menu.js             sync_off.png
```

**Version**

This happens with a self-built Doxygen from Git sources, for either of these revisions (the first is today's HEAD, the second is about 6 months old):
- 1.9.6 (875558cea412de2535b0390c4625bcd0f6bdb411)
- 1.9.5 (86fe8ce908bfd6782633d2d9647f2159c2f08a44)

x86_64 linux:
```
LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: Fedora
Description:    Fedora release 36 (Thirty Six)
Release:        36
Codename:       ThirtySix
```

**Comment by albert-github** [comment]
I can reproduce the problem with the mentioned master version with  a default `Doxyfile`

Problem looks like related to the fact that bar is not documented and this also doesn't give a warning.

**Comment by albert-github** [comment]
@doxygen 
I had a bit a closer look at the problem and the question that arises is: what to do with the items that are not documented? i.e. don't mention them in the index or mention them and create the concept page (like is done with classes).
- In the first case we can probably add a test on `cd->isLinkableInProject()` in `writeConceptRootList` and `writeConceptList` that filters out the non-linkable items (all in `index.cpp`).
- In the second case we should call `writeDocumentation()` also for the non documented concepts, i.e. remove the `cd->isLinkableInProject()` test in `generateConceptDocs` in `doxugen.cpp` but probably also something has to be done with the "conceptbar.html" page regarding where the concept is included from and with concepts nested in other compounds (like namespaces, see attached example. I.e. modify `generateNamespaceConceptDocs` in `doxygen.cpp`).

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/10032163/example.tar.gz)

**Comment by doxygen** [comment]
I've now made the way concepts are handled more in line with classes. The option `HIDE_UNDOC_CLASSES` now also works for concepts. Let me know if you still see issues.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
